File: statistics.ml
let mean lst =
  let sum = List.fold_left ( +. ) 0.0 (List.map float_of_int lst) in
  sum /. float_of_int (List.length lst)

let median lst =
  let sorted = List.sort compare lst in
  let len = List.length sorted in
  if len mod 2 = 1 then
    float_of_int (List.nth sorted (len / 2))
  else
    let mid1 = List.nth sorted ((len / 2) - 1) in
    let mid2 = List.nth sorted (len / 2) in
    (float_of_int (mid1 + mid2)) /. 2.0

let mode lst =
  let counts =
    List.fold_left (fun acc x ->
      let count = try List.assoc x acc with Not_found -> 0 in
      (x, count + 1) :: List.remove_assoc x acc
    ) [] lst in
  let max_freq = List.fold_left (fun acc (_, count) -> max acc count) 0 counts in
  List.fold_left (fun acc (num, count) ->
    if count = max_freq then num :: acc else acc
  ) [] counts

let () =
  let numbers = [1; 2; 2; 3; 4; 4; 4; 5] in
  Printf.printf "Mean: %.2f\n" (mean numbers);
  Printf.printf "Median: %.2f\n" (median numbers);
  Printf.printf "Mode: ";
  List.iter (Printf.printf "%d ") (List.rev (mode numbers));
  Printf.printf "\n"

File: statistics.c
#include <stdio.h>
#include <stdlib.h>

int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}

float mean(int arr[], int size) {
    int sum = 0;
    for(int i = 0; i < size; i++)
        sum += arr[i];
    return (float)sum / size;
}

float median(int arr[], int size) {
    qsort(arr, size, sizeof(int), compare);
    if(size % 2 == 0)
        return (arr[size/2 - 1] + arr[size/2]) / 2.0;
    else
        return arr[size/2];
}

void mode(int arr[], int size) {
    int maxCount = 0;
    printf("Mode: ");
    for(int i = 0; i < size; ++i) {
        int count = 0;
        for(int j = 0; j < size; ++j)
            if(arr[j] == arr[i])
                count++;
        if(count > maxCount)
            maxCount = count;
    }

    for(int i = 0; i < size; ++i) {
        int count = 0;
        for(int j = 0; j < size; ++j)
            if(arr[j] == arr[i])
                count++;
        if(count == maxCount) {
            int isNew = 1;
            for(int k = 0; k < i; ++k)
                if(arr[k] == arr[i])
                    isNew = 0;
            if(isNew)
                printf("%d ", arr[i]);
        }
    }
    printf("\n");
}

int main() {
    int arr[] = {1, 2, 2, 3, 4, 4, 4, 5};
    int size = sizeof(arr)/sizeof(arr[0]);

    printf("Mean: %.2f\n", mean(arr, size));
    printf("Median: %.2f\n", median(arr, size));
    mode(arr, size);

    return 0;
}